pkg/driver/node/mounter/pod_mounter_test.go

```diff
@@ -0,0 +1,379 @@
+	assert.NoError(t, err)
+
+	testCtx.podMounter = podMounter
+
+	return testCtx
+}
+
+func TestPodMounter(t *testing.T) {
+	t.Run("Mounting", func(t *testing.T) {
+		t.Run("Correctly passes mount options", func(t *testing.T) {
+			testCtx := setup(t)
+
+			devNull := mountertest.OpenDevNull(t)
+
+			testCtx.mountSyscall = func(target string, args mountpoint.Args) (fd int, err error) {
+				testCtx.mount.Mount("mountpoint-s3", target, "fuse", nil)
+
+				// Since `PodMounter.Mount` closes the file descriptor once it passes it to Mountpoint,
+				// we should duplicate our file descriptor to ensure underlying file description won't
+				// closed once the file descriptor passed to `PodMounter.Mount` closed.
+				fd, err = syscall.Dup(int(devNull.Fd()))
+				assert.NoError(t, err)
+
+				return fd, nil
+			}
+
+			args := mountpoint.ParseArgs([]string{mountpoint.ArgReadOnly})
+
+			mountRes := make(chan error)
+			go func() {
+				err := testCtx.podMounter.Mount(testCtx.ctx, testCtx.bucketName, testCtx.targetPath, credentialprovider.ProvideContext{
+					AuthenticationSource: credentialprovider.AuthenticationSourceDriver,
+					VolumeID:             testCtx.volumeID,
+					PodID:                testCtx.podUID,
+				}, args)
+				if err != nil {
+					log.Println("Mount failed", err)
+				}
+				mountRes <- err
+			}()
+
+			mpPod := createMountpointPod(testCtx)
+			mpPod.run()
+
+			got := mpPod.receiveMountOptions(testCtx.ctx)
+
+			err := <-mountRes
+			assert.NoError(t, err)
+
+			gotFile := os.NewFile(uintptr(got.Fd), "fd")
+			mountertest.AssertSameFile(t, devNull, gotFile)
+			got.Fd = 0
+
+			assert.Equals(t, mountoptions.Options{
+				BucketName: testCtx.bucketName,
+				Args: []string{
+					"--read-only",
+					"--user-agent-prefix=" + mounter.UserAgent(credentialprovider.AuthenticationSourceDriver, testK8sVersion),
+				},
+				Env: envprovider.Default().List(),
+			}, got)
+		})
+
+		t.Run("Waits for Mountpoint Pod", func(t *testing.T) {
+			testCtx := setup(t)
+
+			go func() {
+				// Add delays to each Mountpoint Pod step
+				time.Sleep(100 * time.Millisecond)
+				mpPod := createMountpointPod(testCtx)
+				time.Sleep(100 * time.Millisecond)
+				mpPod.run()
+				time.Sleep(100 * time.Millisecond)
+				mpPod.receiveMountOptions(testCtx.ctx)
+			}()
+
+			err := testCtx.podMounter.Mount(testCtx.ctx, testCtx.bucketName, testCtx.targetPath, credentialprovider.ProvideContext{
+				VolumeID: testCtx.volumeID,
+				PodID:    testCtx.podUID,
+			}, mountpoint.ParseArgs(nil))
+			assert.NoError(t, err)
+		})
+
+		t.Run("Does not duplicate mounts if target is already mounted", func(t *testing.T) {
+			testCtx := setup(t)
+
+			var mountCount atomic.Int32
+
+			testCtx.mountSyscall = func(target string, args mountpoint.Args) (fd int, err error) {
+				mountCount.Add(1)
+				testCtx.mount.Mount("mountpoint-s3", target, "fuse", nil)
+				return int(mountertest.OpenDevNull(t).Fd()), nil
+			}
+
+			go func() {
```

Comment:
Why are we running this in the background?

Reply:
`mountoptions.Recv` is a blocking operation, this goroutine emulates Mountpoint running in a different process

Reply:
Could we add this as a comment

Reply:
Added `receiveMountOptions`'s comments that this is a blocking operation